app/javascript/packs/pw-strength.js

```diff
@@ -69,6 +70,7 @@ function getFeedback(z) {
     // i18n-tasks-use t('zxcvbn.feedback.this_is_a_top_10_common_password')
     // i18n-tasks-use t('zxcvbn.feedback.this_is_a_very_common_password')
     // i18n-tasks-use t('zxcvbn.feedback.this_is_similar_to_a_commonly_used_password')
+    // i18n-tasks-use t('zxcvbn.feedback.use_a_few_words_avoid_common_phrases')
     // i18n-tasks-use t('zxcvbn.feedback.use_a_longer_keyboard_pattern_with_more_turns')
     return t(`zxcvbn.feedback.${snakeCase(str)}`);
```

Comment:
> This isn't the first time we've had some missing translations with zxcvbn (#5776 comes to mind) is there a better way we can list these, or maybe a safer fallback we can use than to print the slug?

Yeah, these changes should have been included in #5776. As far as preventative measures, I [mentioned at the time](https://github.com/18F/identity-idp/pull/5776#issuecomment-1005072220) that it's a bit hard to test for. There was also a reference to https://github.com/formigarafa/zxcvbn-rb/issues/4 where similar issues had been reported upstream. Since that time, there's been [a comment of all strings](https://github.com/formigarafa/zxcvbn-rb/issues/4#issuecomment-1005585375) we should cross-reference to our translations.

Aside from that, some ideas we could consider:

- Auto-include [all `zxcvbn.feedback` keys](https://github.com/18F/identity-idp/blob/873c71ca8c0cf61d345e0eb9040209802eaefe1d/config/locales/zxcvbn/en.yml#L5-L35) into the locale string extraction
- Don't print original key if there is no translation (at least avoid the non-human-readable text from being shown)

Reply:
I like both those, including all zxcvbn and also not passing through the key (maybe we add a `fallback:` option like ruby i18n has, and we call back to something general like "please use a different password")

Reply:
Another idea is to use TypeScript to enforce coverage for all strings, but the DefinitelyTyped types appear to only [list out the warnings](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/3426972d7780f1688c7e13ce618f59c61d769007/types/zxcvbn/index.d.ts#L109-L125), and not the [suggestions](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/3426972d7780f1688c7e13ce618f59c61d769007/types/zxcvbn/index.d.ts#L102-L106).

Reply:
I double-checked the list in https://github.com/formigarafa/zxcvbn-rb/issues/4#issuecomment-1005585375 and made sure we have all strings listed there. I also added a spec in [6649e74](https://github.com/18F/identity-idp/pull/6008/commits/6649e7441b976d8a2a4b850b67aae3b58894e4be) for the specific behavior being introduced here for an empty password. I was going to add a spec for the joining, though from looking at [the source code](https://github.com/dropbox/zxcvbn/blob/master/src/feedback.coffee), it doesn't appear there's a combination that could result in a multi-suggestion result being used, since we defer to a warning if it's present.

The ideas in https://github.com/18F/identity-idp/pull/6008#discussion_r816132184 are reasonable improvements, but moving forward I might suggest to propose an upstream improvement to the DefinitelyTyped type definitions to include all possible _suggestions_, then update our implementation to strongly type the feedback mappings, e.g.:

```ts
const FEEDBACK: Record<ZXCVBNFeedbackWarning & ZXCVBNFeedbackSuggestion, string> = {
  "Straight rows of keys are easy to guess": '...',
  // ...
};
```